remote/RemoteImGuiServer.cpp

```diff
@@ -31,93 +32,88 @@ namespace imgui {
-					_onImMouseWheelDelta(mouseWheelDelta);
-				}
-			}
-			else if (strstr((char *)data, "ImKeyDown")) {
-				int key, shift, ctrl;
-				if (sscanf((char *)data, "ImKeyDown=%d,%d,%d", &key, &shift, &ctrl) == 3) {
-					_onImKeyDown(key, shift, ctrl);
+	void RemoteImGuiServer::OnMessage(IWebSocketServer::OpCode opcode, const void *data, int size) {
+		switch (opcode) {
+			case IWebSocketServer::Text: {
+				if (size > 0) {
+					const unsigned char * text = static_cast<const unsigned char *>(data);
+					RemoteMessageType messageType = static_cast<RemoteMessageType>(text[0]);
+					_handleMessage(messageType, &text[1], size - 1);
 				}
+				break;
 			}
-			else if (strstr((char *)data, "ImKeyUp")) {
-				int key;
-				if (sscanf((char *)data, "ImKeyUp=%d", &key) == 1) {
-					_onImKeyUp(key);
-				}
+			case IWebSocketServer::Disconnect: {
+				mIsClientActive = false;
+				break;
 			}
-			else if (strstr((char *)data, "ImKeyPress")) {
-				unsigned int key;
-				if (sscanf((char *)data, "ImKeyPress=%d", &key) == 1)
-					_onImKeyPress(key);
+			case IWebSocketServer::Continuation:
+			case IWebSocketServer::Binary:
+			case IWebSocketServer::Ping:
+			case IWebSocketServer::Pong: {
+				// Valid, but unhandled
+				break;
 			}
-			else if (strstr((char *)data, "ImClipboard=")) {
-				_onImClipboard(nullptr, &((char *)data)[strlen("ImClipboard=")]);
+			default: {
+				// Unsupported message type
+				assert(0);
+				break;
 			}
-			break;
-
-		
-			break;
-
-		case IWebSocketServer::Disconnect:
-			mIsClientActive = false;
-			break;
-
-		case IWebSocketServer::Binary:
-		case IWebSocketServer::Ping:
-		case IWebSocketServer::Pong:
-			break;
-		
-		default:
-			assert(0);
-			break;
 		}
 	}
 
+	bool RemoteImGuiServer::_getIsActive() const {
+		return mIsClientActive;
+	}
+
 	void RemoteImGuiServer::_sendFrame(const Frame& frame) {
 		SendBinary(frame.data, frame.size);
 	}
 
-	bool RemoteImGuiServer::_getIsActive() const {
-		return mIsClientActive;
+	void RemoteImGuiServer::_handleMessage(RemoteMessageType messageType, const void * data, int size) {
+		switch (messageType) {
+			case RemoteMessageType::RelayRoomJoined:
+			case RemoteMessageType::RelayRoomUpdate: {
+				// Valid, but unhandled
+				break;
+			}
+			case RemoteMessageType::ImInit: {
+				// If not active, don't process input
+				if (!_getIsActive()) {
+					mIsClientActive = true;
+
+					// ImGUI initialization request received - send init data
+					SendText(&messageType, 1);
+
+					// Send font texture
+					_sendFontFrame();
+				}
+				break;
+			}
+			case RemoteMessageType::ImMouseMove:
+			case RemoteMessageType::ImMousePress:
+			case RemoteMessageType::ImMouseWheelDelta:
+			case RemoteMessageType::ImKeyDown:
+			case RemoteMessageType::ImKeyUp:
+			case RemoteMessageType::ImKeyPress:
+			case RemoteMessageType::ImClipboard: {
+				// Handling other messages is not specific to relay servers
```

Comment:
>/ Handling other messages is not specific to relay servers [](start = 5, length = 58)

We probably shouldn't mention anything about Relay in this class since it it totally stand-alone